std/format.d

```diff
@@ -867,7 +867,7 @@ struct FormatSpec(Char)
                 // Get the matching balanced paren
                 for (uint innerParens;;)
                 {
-                    enforce(j < trailing.length,
+                    enforce(j + 1 < trailing.length,
```

Comment:
Maybe this should throw `FormatException`, not `Exception`?

Description of `FormatException`:
> Signals a mismatch between a format and its corresponding argument.

Not quite what is going on here... maybe a new exception type for misformed format strings?